[DOC] Remove `v1` API conversion documentation (#12695)
strimzi/strimzi-kafka-operator: documentation/modules/upgrading/con-api-conversion-v1.adoc

// assembly-upgrade.adoc

[id='con-api-conversion-v1-{context}']
= Upgrading from versions earlier than 1.0.0

[role="_abstract"]

> When upgrading from a version prior to Strimzi 1.0.0, you **must** convert your custom resources and CRDs to the `v1` API.

Reviewer: ```suggestion
When upgrading from a version earlier than Strimzi 1.0.0, you **must** convert your custom resources and CRDs to the `v1` API.
```

> Please follow the link:https://strimzi.io/docs/operators/1.0.0/deploying.html#assembly-api-conversion-str[Strimzi 1.0.0 documentation^] for more details.

Reviewer: ```suggestion
For guidance on the conversion, see the link:https://strimzi.io/docs/operators/1.0.0/deploying.html#assembly-api-conversion-str[Strimzi 1.0.0 documentation^].
```
